packages/jest-snapshot/src/State.js

```diff
@@ -83,7 +83,7 @@ class SnapshotState {
 
     const isEmpty = Object.keys(this._snapshotData).length === 0;
 
-    if ((this._dirty || this._uncheckedKeys.size) && !isEmpty) {
+    if ((this._dirty || this._uncheckedKeys.size || update) && !isEmpty) {
```

Comment:
Can we make this so we only write the file if update is given & the snapshot header is either missing or there is a mismatch? I wanna prevent writes from happening if we don't need them.